### DISCUSSION TITLE
Week 5

### DISCUSSION THREAD

**Discussion by Stev3eM** [SE-TINF22B6/CookHub#45] [Show and tell]
Hello backpackers,

This is our weekly discussion-blog to keep you up to date within our project progression.

For week 5 we have another view updates for you:

✅ We implemented our Testing-Frameworks:
 - Backend: xUnit.NET
 - Frontend: Jest für Frontend

✅ Implemented CI workflow:
 - with every pull request and merge:
     - the **dependencies** of our project get restored/updated (`npm ci`, `dotnet restore`)
     - a **build** of our app is automatically executed on a github server (`npm run build`, `dotnet build`)
     - all **tests** are run to check whether the commit might break the app (`npm test`, `dotnet test`)
 - if one of the actions fails, we are also get notified by email and pull requests that caused the action to fail are blocked
 - We also tested it

✅ CI Setup:
 - Here is our [config file backend action](https://github.com/SE-TINF22B6/CookHub/blob/main/.github/workflows/dotnet.yml)
 - Here is our  [config file frontend action](https://github.com/SE-TINF22B6/CookHub/blob/main/.github/workflows/node.js.yml)
- Here is the [Readme](https://github.com/SE-TINF22B6/CookHub/blob/main/.github/workflows/donet.yml_Readme) to the configs

See you soon with another update!

Your CookHub Team
<img width="50" alt="image" src="https://github.com/SE-TINF22B6/CookHub/assets/113697513/cdbd0688-3eb9-46fa-a608-baf2efbaba88">

**Comment by BenjaminB109** [comment]
Dear CookHub Team,
I am glad that it worked to implement the CI workflow. One possible suggestion for improvement could be to specify the type of notifications. For example, you could specify which tests have failed or which aspects of the build have been affected. Keep it up!

**Comment by fehlend0** [comment]
Dear CookHub Team,

Using xUnit.NET for the backend and Jest for the frontend is to our mind a good approach. As for your CI workflows implementation, the automatic execution of builds and tests on a GitHub server, along with email notifications for failed actions, is a robust setup. Great, that you've tested it, it adds an extra layer of reliability. 

All in all, you’ve done a great job, keep us informed about your future progress!

Your Spotify Team
